File: Programmers/1.py
def solution(board, moves):
    answer = 0
    
    bk = [0]
    
    for value in moves:
        while True:
            if not board[value-1]:
                break
            temp = board[value-1].pop()
            if temp != 0:
                if bk[-1] == temp:
                    del bk[-1]
                    answer += 2
                else:
                    bk.append(temp)
                break
                
    #print(board)
    return answer

File: Programmers/1.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> board,vector <int> moves) {
    
    int answer = 0;
    int temp;
    vector<int> bk;
    
    bk.push_back(0);
    
    for(int i = 0; i < moves.size(); i++)
    {           
        for( ;  0 == (board[moves[i]-1].empty()) ;)
        {
            temp = board[moves[i]-1].back();
            board[moves[i]-1].pop_back();
            if(temp)
            {
                if(temp == bk.back())
                {
                    bk.pop_back();
                    answer +=2;
                }    
                else
                {
                    bk.push_back(temp);
                }
                
                cout << bk;
                break;
            }
        }    
    }
    return answer;
}